opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -51,12 +51,47 @@ static char *ofi_progress_mode;
 static bool disable_sep;
 static int mca_btl_ofi_init_device(struct fi_info *info);
 
+static int
+is_in_list(char **list, char *item)
+{
+    int i = 0;
+
+    if ((NULL == list) || (NULL == item)) {
+        return 0;
+    }
+
+    while (NULL != list[i]) {
+        if (0 == strncasecmp(item, list[i], strlen(list[i]))) {
+            return 1;
+        } else {
+            i++;
+        }
+    }
+
+    return 0;
+}
+
 /* validate information returned from fi_getinfo().
  * return OPAL_ERROR if we dont have what we need. */
-static int validate_info(struct fi_info *info, uint64_t required_caps)
+static int validate_info(struct fi_info *info, uint64_t required_caps,
+                         char **include_list, char **exclude_list)
 {
     int mr_mode;
 
+    if (NULL != include_list && !is_in_list(include_list, info->fabric_attr->prov_name)) {
+        opal_output_verbose(1, opal_common_ofi.output,
+                            "%s:%d: btl:ofi: \"%s\" not in include list\n",
+                            __FILE__, __LINE__,
+                            info->fabric_attr->prov_name);
+        return OPAL_ERROR;
+    } else if (NULL != exclude_list && is_in_list(exclude_list, info->fabric_attr->prov_name)) {
```

Comment:
Where does info->fabric_attr->prov_name get set now? I saw that it's no longer set directly in mca_btl_ofi_component_init . Does it get set somewhere else?

Reply:
Looks like the OFI BTL doesn't use that attribution in the info arg to fi_getinfo with these changes.  I think that's okay, we're now just filtering what fi_getinfo returns for provider info and only picking the one that fits with the specified include/exclude provider lists.

Reply:
Okay. Thank you for clarifying, Howard!